scripts/tslint/rules/align2Rule.ts

```diff
@@ -0,0 +1,97 @@
+// Based on https://github.com/palantir/tslint/blob/master/src/rules/alignRule.ts
+// Has code to count a comment at the beginning of a line as the start of that line
+
+import * as ts from "typescript";
+
+import * as Lint from "tslint";
+
+export class Rule extends Lint.Rules.AbstractRule {
+    public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+        return this.applyWithWalker(new AlignWalker(sourceFile, this.ruleName, undefined));
+    }
+}
+
+class AlignWalker extends Lint.AbstractWalker<void> {
+    public walk(sourceFile: ts.SourceFile) {
+        const cb = (node: ts.Node): void => {
+            switch (node.kind) {
+                case ts.SyntaxKind.SourceFile:
+                case ts.SyntaxKind.ModuleBlock:
+                case ts.SyntaxKind.Block:
+                case ts.SyntaxKind.CaseClause:
+                case ts.SyntaxKind.DefaultClause:
+                    this.checkAlignment((node as ts.SourceFile | ts.ModuleBlock | ts.Block | ts.CaseClause | ts.DefaultClause).statements.filter((s) => s.kind !== ts.SyntaxKind.EmptyStatement));
+                    break;
+                case ts.SyntaxKind.ArrayLiteralExpression:
+                case ts.SyntaxKind.ArrayBindingPattern:
+                    this.checkAlignment((node as ts.ArrayBindingOrAssignmentPattern).elements);
+                    break;
+                case ts.SyntaxKind.TupleType:
+                    this.checkAlignment((node as ts.TupleTypeNode).elementTypes);
+                    break;
+                case ts.SyntaxKind.ObjectLiteralExpression:
+                    this.checkAlignment((node as ts.ObjectLiteralExpression).properties);
+                    break;
+                case ts.SyntaxKind.ObjectBindingPattern:
+                    this.checkAlignment((node as ts.ObjectBindingPattern).elements);
+                    break;
+                case ts.SyntaxKind.ClassDeclaration:
+                case ts.SyntaxKind.ClassExpression:
+                    this.checkAlignment((node as ts.ClassLikeDeclaration).members.filter((m) => m.kind !== ts.SyntaxKind.SemicolonClassElement));
+                    break;
+                case ts.SyntaxKind.InterfaceDeclaration:
+                case ts.SyntaxKind.TypeLiteral:
+                    this.checkAlignment((node as ts.InterfaceDeclaration | ts.TypeLiteralNode).members);
+            }
+            return ts.forEachChild(node, cb);
+        };
+        return cb(sourceFile);
+    }
+
+    private checkAlignment(nodes: ReadonlyArray<ts.Node>) {
+        if (nodes.length <= 1) {
+            return;
+        }
+        const sourceFile = this.sourceFile;
+
+        let pos = this.getStart(nodes[0]).lineAndChar;
+        const alignToColumn = pos.character;

```

Comment:
`alignToColumn` should _probably_ be locked to a multiple of the indentation size.